src/main/toggles.ts

```diff
@@ -13,36 +14,37 @@ import { memoize, objectMatchesSchema } from './utils'
 
 const rawToggles: ToggleMap = new Map()
 
-const lazyToggles: () => Promise<ToggleMap> = memoize(() => {
+const lazyToggles: (togglePath: string) => Promise<ToggleMap> = memoize<string, Promise<ToggleMap>>((togglePath: string) => {
     return new Promise((resolve, reject) => {
-        config().watch<Array<IToggleDescription>>(DEFAULT_TOGGLES_PATH).onValue((toggles): void => {
-            if (objectMatchesSchema(toggleSchema, { toggles })) {
-                toggles.forEach((next: IToggleDescription) => {
-                    rawToggles.set(next.id, next)
+        config().watch<IToggleDescription>(togglePath).onValue(({toggles}): void => {
+            if (objectMatchesSchema(toggleSchema, toggles)) {
+                Object.keys(toggles).forEach((key) => {
+                    if (toggles[key]) {
+                        rawToggles.set(key, toggles[key])
+                    }
                 })
+                resolve(Promise.resolve(rawToggles))
```

Comment:
There seems to be an extra resolve here.

Reply:
👍